### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by MattWellie**
Adds documentation relating to the recently transplanted code (git.py from analysis-runner, [link](https://github.com/populationgenomics/analysis-runner/blob/main/analysis_runner/git.py)). Eventually this code could be removed from analysis runner and only hosted here, so user understanding of this will be important.

- includes code examples for new methods in a collection of .MD files
- renames `hail.py` to `hail_batch.py` (this can be reverted, breaking change may not be worth pleasing an overzealous linter)
- new method(s) for identifying the organisation from the repository path
  - available now that we are not hard-restricting to `populationgenomics` 

[x] linter checks pass

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: Lint
-on: [push, pull_request]
+on: [push]
```
Comment (lgruen) [inline .github/workflows/lint.yaml:2]: Will this also still run the linter when a new PR is created?

File: documentation/hail_batch.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+def main():
+    # initiate a hail query back end
+    init_batch()
+
+    # hail object methods
+    ...
+```
+
+When running a Hail Batch workflow using the analysis-runner, the core Driver image will
+have authentication set up, but separate child batches will start with fresh containers.
```
Comment (lgruen) [inline documentation/hail_batch.md]:
```suggestion
have authentication set up, but separate child batches will start with fresh jobs.
```

**Review by lgruen — APPROVED** [review]
Thank you!

**Comment by lgruen** [comment]
Due to the module being renamed, this should be released with a major version bump.

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (MattWellie) [reply .github/workflows/lint.yaml:2]: yes and no?! There is currently redundancy as the linter runs when you push, then runs again when you make a PR from that push. If you push and it fails, the PR will still show that latest commit as a CI fail without rerunning

**Comment by MattWellie** [comment]
Updated with a major bump, but not releasing yet as unpinned builds will break

File: documentation/hail_batch.md (hunk 2 above)
Comment (MattWellie) [reply documentation/hail_batch.md]: I don't think either of these is the exact right message. The message I intended was that although the Driver container has authentication, the same auth also needs to be triggered within containers used for each batch job. E.g. you can run a `gsutil cp` in the driver image, but shouldn't expect that to work in a job container, unless you trigger auth there as well.

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (illusional) [reply .github/workflows/lint.yaml:2]: Anecdotally I saw no real changed behaviour when I removed it for sample-metadata.

File: cpg_utils/hail_batch.py [hunk 3]
```diff
@@ -30,7 +36,7 @@ def init_batch(**kwargs):
     )
 
 
-def copy_common_env(job: hb.job.Job) -> None:
+def copy_common_env(job: Union[hb.batch.job.BashJob, hb.batch.job.Job]) -> None:
```
Comment (illusional) [inline cpg_utils/hail_batch.py]: BashJob is a type of Job right?

**Review by illusional — APPROVED** [review]

File: documentation/hail_batch.md (hunk 2 above)
Comment (lgruen) [reply documentation/hail_batch.md]: Yes, I think that's clear from the current text, thanks!

File: cpg_utils/hail_batch.py (hunk 3 above)
Comment (MattWellie) [reply cpg_utils/hail_batch.py]: Yeah, this was a silly change. The type reference for Job was wrong, but doesn't need to be duplicated for subtypes. Altered in latest commit

**Pull request merged**